### PR TITLE
Feature/299

### PR DESCRIPTION
**Pull request by TreeMama**
@halmueller can you give us a PR :)

### CODE REVIEW

File: ios/iSeaTree/AppDelegate.m [hunk 1]
```diff
@@ -0,0 +1,104 @@
+#import "AppDelegate.h"
+
+#import <React/RCTBridge.h>
+#import <React/RCTBundleURLProvider.h>
+#import <React/RCTRootView.h>
+
+#import <Firebase.h>
```
Comment (halmueller) [inline ios/iSeaTree/AppDelegate.m]: This #import is failing. I think it's a Cocoapods issue. I'll work it from here.

**Review by halmueller — COMMENTED** [review]
Note: do not merge yet. It doesn't build for iOS.

@monikap0612 (or anyone else): was there a particular reason you chose Objective-C instead of Swift for the AppDelegate/initialization?

Also: this merge request is huge. It's way too big for anyone to understand who didn't write it. Please make smaller commits/merges, and make them single topic. The camera work and the Expo migration are intertwined, but they're unrelated and should have gone as separate merges.

File: ios/Podfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  # Enables Flipper.
+  #
+  # Note that if you have use_frameworks! enabled, Flipper will not work and
+  # you should disable these next few lines.
+
+  # use_flipper!
+  # use_flipper!({ 'Flipper' => '0.74.0' })
+  # post_install do |installer|
+  #   flipper_post_install(installer)
+  # end
```
Comment (halmueller) [inline ios/Podfile]: Why is this code commented out and not simply deleted? Please delete it or add a comment explaining why it's better to leave it in. Commented-out code is a "code smell".

**Comment by TreeMama** [comment]
> Note: do not merge yet. It doesn't build for iOS.
> 
> @monikap0612 (or anyone else): was there a particular reason you chose Objective-C instead of Swift for the AppDelegate/initialization?
> 

I don't know if Monika told you - but she said when I posed your question:

"There is no particular reason to choose Objective c, when we create a new project, it will automatically create project in Objective-C"

Regarding the build issues she suggests:

"Regarding iOS issue, pls delete pod.lock file & try to do below command once again:

- cd ios
    - pod install
    - cd ..
    - npx react-native run-ios
    
    "

Does that help @halmueller ?

**Comment by halmueller** [comment]
It's missing some cocopods. Fixed that, but uncovered a regression (the premature permission request). I'm looking for the original issue and fix. Looks like PR https://github.com/TreeMama/iSeaTree-React-Prototype/pull/228 introduced the regression.

**Pull request merged**